### PR TITLE
Birthday Calendar Reminder Setting

### PR DESCRIPTION
**Pull request by cneukom**
First part for closing https://github.com/nextcloud/server/issues/1505.

### CODE REVIEW

File: apps/dav/lib/CalDAV/BirthdayService.php [hunk 1]
```diff
@@ -98,9 +98,11 @@
 				continue;
 			}
 
+			$reminderOffset = $this->getReminderOffsetForUser($principalUri);
+
 			$calendar = $this->ensureCalendarExists($principalUri);
 			foreach ($datesToSync as $type) {
-				$this->updateCalendar($cardUri, $cardData, $book, (int) $calendar['id'], $type);
+				$this->updateCalendar($cardUri, $cardData, $book, (int) $calendar['id'], $type, $reminderOffset);
```
[the remark this answers is not in the document]
Comment (cneukom) [reply apps/dav/lib/CalDAV/BirthdayService.php:112]: Fixed in 9eb276b20b051c31b349671600961c60d94ad175

[the remark this answers is not in the document]
Comment (cneukom) [reply apps/dav/lib/CalDAV/BirthdayService.php:112]: Fixed in 9eb276b20b051c31b349671600961c60d94ad175

[the remark this answers is not in the document]
Comment (cneukom) [reply apps/dav/lib/CalDAV/BirthdayService.php:112]: Fixed in 9eb276b20b051c31b349671600961c60d94ad175

**Comment by miaulalala** [comment]
Performance check fails bc remote is not origin.

File: apps/dav/lib/CalDAV/BirthdayService.php [hunk 2]
```diff
@@ -341,16 +354,18 @@ protected function getAllAffectedPrincipals(int $addressBookId) {
...
 	 * @throws InvalidDataException
 	 * @throws \Sabre\DAV\Exception\BadRequest
 	 */
 	private function updateCalendar(string $cardUri,
 									string $cardData,
 									array $book,
 									int $calendarId,
-									array $type):void {
+									array $type,
+									?string $reminderOffset):void {
```
Comment (miaulalala) [inline apps/dav/lib/CalDAV/BirthdayService.php:366]: please initialise this with `null` just in case any other app uses the method.

**Review by miaulalala — CHANGES_REQUESTED** [review]

File: apps/dav/lib/CalDAV/BirthdayService.php (hunk 2 above)
Comment (tcitworld) [reply apps/dav/lib/CalDAV/BirthdayService.php:366]: It's a private method :)

**Review by miaulalala — APPROVED** [review]
LGTM

**Review by tcitworld — APPROVED** [review]

**Pull request merged**